framework/branch-key-store.md

```diff
@@ -520,10 +632,24 @@ except the logical table name
 MUST equal the value with the same key in the AWS DDB response item.
 The key `enc` MUST NOT exist in the constructed [encryption context](#encryption-context).
 
+If the Keystore's [AWS KMS Configuration](#aws-kms-configuration) is `KMS Key ARN` or `KMS MRKey ARN`,
+the `kms-arn` field of the DDB response item MUST be
+[compatible with](#aws-key-arn-compatibility) the configured `KMS ARN` in
+the [AWS KMS Configuration](#aws-kms-configuration) for this keystore,
+or the operation MUST fail.
+
+If the Keystore's [AWS KMS Configuration](#aws-kms-configuration) is `Discovery` or `MRDiscovery`,
+the `kms-arn` field of DDB response item MUST NOT be an Alias
+or the operation MUST fail.
+
+<!--  "For all Branch Keys created by any version of the MPL, an Alias for kms-arn is impossible." -->
+
 When calling [AWS KMS Decrypt](https://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html),
 the keystore operation MUST call with a request constructed as follows:
 
-- `KeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
+- `KeyId`, if the KMS Configuration is Discovery, MUST be the `kms-arn` attribute value of the AWS DDB response item.
+  If the KMS Configuration is MRDiscovery, `KeyId` MUST be the `kms-arn` attribute value of the AWS DDB response item, with the region replaced by the configured region.
+  Otherwise, it MUST BE the Keystore's `KMS ARN`.
```

Comment:
This is a mouthful. Can we simplify this?

Reply:
I am open to suggestions, but, at this time, looking to ship software, and do not have the capacity to refactor this.